apps/web/src/scenes/UserGalleryPage/UserSharedFollowers.tsx

```diff
@@ -0,0 +1,130 @@
+import { useCallback, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import InteractiveLink from '~/components/core/InteractiveLink/InteractiveLink';
+import { HStack } from '~/components/core/Spacer/Stack';
+import { BaseS } from '~/components/core/Text/Text';
+import { useTrack } from '~/contexts/analytics/AnalyticsContext';
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { UserSharedFollowersFragment$key } from '~/generated/UserSharedFollowersFragment.graphql';
+import { useIsMobileWindowWidth } from '~/hooks/useWindowSize';
+
+import PaginatedUsersList from '../Modals/PaginatedList/PaginatedUsersList';
+
+type Props = {
+  queryRef: UserSharedFollowersFragment$key;
+};
+
+export default function UserSharedFollowers({ queryRef }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment UserSharedFollowersFragment on GalleryUser {
+        __typename
+        sharedFollowers(first: $sharedFollowersFirst, after: $sharedFollowersAfter)
+          @connection(key: "UserSharedInfoFragment_sharedFollowers") {
+          edges {
+            node {
+              __typename
+              ... on GalleryUser {
+                __typename
+                username
+              }
+            }
+          }
+          pageInfo {
+            total
+          }
+        }
+        ...PaginatedUsersListFragment
+      }
+    `,
+    queryRef
+  );
+
+  const { showModal } = useModalActions();
+  const track = useTrack();
+  const isMobile = useIsMobileWindowWidth();
+
+  const handleShowAllFollowersClick = useCallback(() => {
+    track('User Gallery - Show All Shared Followers Click');
+    showModal({
+      content: <PaginatedUsersList queryRef={query} />,
+      headerText: 'Followers you know',
+      isPaddingDisabled: true,
+      isFullPage: isMobile,
+    });
+  }, [isMobile, query, showModal, track]);
+
+  const sharedFollowers = useMemo(
+    () => query.sharedFollowers?.edges?.map((edge) => edge?.node) ?? [],
+    [query.sharedFollowers?.edges]
+  );
+
+  const totalSharedFollowers = query.sharedFollowers?.pageInfo?.total ?? 0;
+
+  // Determine how many users to display by username
+  const followersToDisplay = useMemo(() => {
+    // In most cases we display a max of 2 usernames. ie "username1, username2 and 3 others you follow"
+    // But if there are exactly 3 shared followers, we display all 3 usernames. ie "username1, username2 and username3"
+    const maxUsernamesToDisplay = totalSharedFollowers === 3 ? 3 : 2;
+    return sharedFollowers.slice(0, maxUsernamesToDisplay);
+  }, [sharedFollowers, totalSharedFollowers]);
+
+  const content = useMemo(() => {
+    // Display up to 3 usernames
+    const result = followersToDisplay.map(
+      (user) =>
+        user && (
```

Comment:
I feel like some of this code might be more logically correct if we `removeNullValues` before we `slice` above. That way we know for sure that we're actually displaying the number of users we think we are. 

I would add it right before the slice in the `sharedFollowers` `useMemo` bit. Then, all of your downstream usages shouldn't have to deal with the null values.

Reply:
good point - I added `removeNullValues` to the `sharedCommunities` memo

Reply:
good point - I added removeNullValues to the sharedCommunities memo